utils/tower/src/middleware.rs

```diff
@@ -68,40 +65,69 @@ where
 
 impl<B> Default for CountBytesBody<B>
 where
-    B: HttpBody<Data = Bytes> + Default,
+    B: Body<Data = Bytes> + Default,
 {
     fn default() -> Self {
         Self { inner: Default::default(), counter: Default::default() }
     }
 }
 
+pin_project! {
+    pub struct ChannelBody<T> {
+        #[pin]
+        rx: tokio::sync::mpsc::UnboundedReceiver<Frame<T>>,
+    }
+}
+
+impl<T> ChannelBody<T> {
+    pub fn new() -> (tokio::sync::mpsc::UnboundedSender<Frame<T>>, Self) {
+        let (tx, rx) = tokio::sync::mpsc::unbounded_channel();
```

Comment:
Can you explain the possible implications of using an unbounded channel here?

Reply:
In the original implementation it was bounded channel with 0-sized buffer, meaning it was a blocking operation. It had inconsistent api between hyper channel and tokio::mpsc channel, where 0 is invalid value. I believe it should be bounded channel of size equal to 1.


Answering the question: unbounded channel can lead to bugger memory allocations if we don't have enough tokio workers to read from the channel.

I will change it to bounded(1), wdyt?